ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/QueryAnonymizer.java

```diff
@@ -887,15 +887,31 @@ private static String visitWithinExpression(final WithinExpressionContext contex
         final String after =
             getSizeAndUnitFromJoinWindowSize(beforeAndAfterJoinWindow.joinWindowSize(1));
         stringBuilder.append(String.format("(%s, %s)", before, after));
+      } else if (context instanceof SqlBaseParser.JoinWindowWithGraceContext) {
+        final SqlBaseParser.JoinWindowWithGraceContext joinWithGracePeriod =
+            (SqlBaseParser.JoinWindowWithGraceContext) context;
+
+        stringBuilder.append(anonymizeJoinWindowWithGrace(joinWithGracePeriod));
       } else {
         throw new RuntimeException("Expecting either a single join window, ie \"WITHIN 10 "
-            + "seconds\", or a join window with before and after specified, "
-            + "ie. \"WITHIN (10 seconds, 20 seconds)");
+            + "seconds\", a join window with before and after specified, "
+            + "ie. \"WITHIN (10 seconds, 20 seconds)\", or a join window with a grace period, "
+            + "ie. \"WITHIN (SIZE 10 seconds, GRACE PERIOD 5 seconds)\"");
       }
 
       return stringBuilder.toString();
     }
 
+    private static String anonymizeJoinWindowWithGrace(
+        final SqlBaseParser.JoinWindowWithGraceContext joinWindowWithGrace
+    ) {
+      return String.format("(SIZE %s %s, GRACE PERIOD %s %s) ",
+          "'0'",
+          joinWindowWithGrace.joinWindowSize().windowUnit().getText().toUpperCase(),
+          "'0'",
+          joinWindowWithGrace.gracePeriodClause().windowUnit().getText().toUpperCase());
+    }
+
     private static String getSizeAndUnitFromJoinWindowSize(
```

Comment:
Should we rename this method to `anonymizeSizeAndUnitFromJoinWindowSize` (also, should we remove `AndUnit` from the name, as we preserve the units? Or is the method incorrect and it _should_ indeed anonymize the unit, too)?

Reply:
I just renamed to `anonymizeJoinWindowSize`. The anonymizer is not changing the unit so I removed it.